BUG: groupby .agg('size') concat issue after execute map

Note that the issue tracker is NOT the place for general support.  For
discussions about development, questions about usage, or any general questions,
contact us on https://discuss.xorbits.io/.

We did not meet this issue in our regular CI, because the CI environment is either a pure CPU environment or only the GPU UT is tested.

If you run ``test_dataframe_groupby_agg`` this UT in a GPU env, the error is shown below:
```
Traceback (most recent call last):
  File "/tmp/pycharm_project_972/python/xorbits/_mars/services/subtask/worker/processor.py", line 212, in _execute_operand
    return execute(ctx, op)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/core/operand/core.py", line 492, in execute
    result = executor(results, op)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 344, in execute
    ctx[chunk.key] = tuple(
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 345, in <genexpr>
    _base_concat(chunk, [input[i] for input in inputs])
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 254, in _base_concat
    return _auto_concat_series_chunks(chunk, inputs)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 316, in _auto_concat_series_chunks
    concat = xdf.concat(inputs)
  File "/home/lichengjie/miniconda3/lib/python3.9/site-packages/cudf/core/reshape.py", line 294, in concat
    raise TypeError(f"cannot concatenate object of type {type(o)}")
TypeError: cannot concatenate object of type <class 'pandas.core.frame.DataFrame'>
ERROR    xorbits._mars.services.scheduling.worker.execution:execution.py:133 Failed to run subtask s6uaKkROrqkjN6Oe2YOEA09v on band numa-0
Traceback (most recent call last):
  File "/tmp/pycharm_project_972/python/xorbits/_mars/services/subtask/worker/processor.py", line 212, in _execute_operand
    return execute(ctx, op)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/core/operand/core.py", line 492, in execute
    result = executor(results, op)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 344, in execute
    ctx[chunk.key] = tuple(
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 345, in <genexpr>
    _base_concat(chunk, [input[i] for input in inputs])
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 254, in _base_concat
    return _auto_concat_series_chunks(chunk, inputs)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 316, in _auto_concat_series_chunks
    concat = xdf.concat(inputs)
  File "/home/lichengjie/miniconda3/lib/python3.9/site-packages/cudf/core/reshape.py", line 294, in concat
    raise TypeError(f"cannot concatenate object of type {type(o)}")
TypeError: cannot concatenate object of type <class 'pandas.core.frame.DataFrame'>
ERROR    xorbits._mars.services.task.execution.mars.stage:stage.py:169 Subtask lwr7brX47wbtKTk4g1RvTeQo errored
Traceback (most recent call last):
  File "/tmp/pycharm_project_972/python/xorbits/_mars/services/subtask/worker/processor.py", line 212, in _execute_operand
    return execute(ctx, op)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/core/operand/core.py", line 492, in execute
    result = executor(results, op)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 344, in execute
    ctx[chunk.key] = tuple(
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 345, in <genexpr>
    _base_concat(chunk, [input[i] for input in inputs])
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 254, in _base_concat
    return _auto_concat_series_chunks(chunk, inputs)
  File "/tmp/pycharm_project_972/python/xorbits/_mars/dataframe/merge/concat.py", line 316, in _auto_concat_series_chunks
    concat = xdf.concat(inputs)
  File "/home/lichengjie/miniconda3/lib/python3.9/site-packages/cudf/core/reshape.py", line 294, in concat
    raise TypeError(f"cannot concatenate object of type {type(o)}")
TypeError: cannot concatenate object of type <class 'pandas.core.frame.DataFrame'>

xorbits/_mars/dataframe/groupby/tests/test_groupby_execution.py:327 (test_dataframe_groupby_agg)
setup = <xorbits._mars.deploy.oscar.session.SyncSession object at 0x7f5360bfa0d0>

    def test_dataframe_groupby_agg(setup):
        agg_funs = [
            "std",
            "mean",
            "var",
            "max",
            "count",
            "size",
            "all",
            "any",
            "skew",
            "kurt",
            "sem",
            "nunique",
        ]

        rs = np.random.RandomState(0)
        raw = pd.DataFrame(
            {
                "c1": np.arange(100).astype(np.int64),
                "c2": rs.choice(["a", "b", "c"], (100,)),
                "c3": rs.rand(100),
            }
        )
        mdf = md.DataFrame(raw, chunk_size=13)

        for method in ["tree", "shuffle"]:
            for sort in [True, False]:
                r = mdf.groupby("c2").agg("size", method=method)
                pd.testing.assert_series_equal(
>                   r.execute().fetch().sort_index(),
                    raw.groupby("c2").agg("size").sort_index(),
                )

xorbits/_mars/dataframe/groupby/tests/test_groupby_execution.py:358: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
xorbits/_mars/core/entity/tileables.py:405: in execute
    result = self.data.execute(session=session, **kw)
xorbits/_mars/core/entity/executable.py:152: in execute
    return execute(self, session=session, **kw)
xorbits/_mars/deploy/oscar/session.py:1709: in execute
    return session.execute(
xorbits/_mars/deploy/oscar/session.py:1526: in execute
    execution_info: ExecutionInfo = fut.result(
/home/lichengjie/miniconda3/lib/python3.9/concurrent/futures/_base.py:446: in result
    return self.__get_result()
/home/lichengjie/miniconda3/lib/python3.9/concurrent/futures/_base.py:391: in __get_result
    raise self._exception
xorbits/_mars/deploy/oscar/session.py:1689: in _execute
    await execution_info
/home/lichengjie/miniconda3/lib/python3.9/asyncio/tasks.py:688: in _wrap_awaitable
    return (yield from awaitable.__await__())
xorbits/_mars/deploy/oscar/session.py:102: in wait
    return await self._aio_task
xorbits/_mars/deploy/oscar/session.py:848: in _run_in_background
    raise task_result.error.with_traceback(task_result.traceback)
xorbits/_mars/services/task/supervisor/processor.py:388: in run
    await self._process_stage_chunk_graph(*stage_args)
xorbits/_mars/services/task/supervisor/processor.py:265: in _process_stage_chunk_graph
    chunk_to_result = await self._executor.execute_subtask_graph(
xorbits/_mars/services/task/execution/mars/executor.py:203: in execute_subtask_graph
    return await stage_processor.run()
xorbits/_mars/services/task/execution/mars/stage.py:233: in run
    return await self._run()
xorbits/_mars/services/task/execution/mars/stage.py:253: in _run
    raise self.result.error.with_traceback(self.result.traceback)
xorbits/_mars/services/subtask/worker/processor.py:212: in _execute_operand
    return execute(ctx, op)
xorbits/_mars/core/operand/core.py:492: in execute
    result = executor(results, op)
xorbits/_mars/dataframe/merge/concat.py:344: in execute
    ctx[chunk.key] = tuple(
xorbits/_mars/dataframe/merge/concat.py:345: in <genexpr>
    _base_concat(chunk, [input[i] for input in inputs])
xorbits/_mars/dataframe/merge/concat.py:254: in _base_concat
    return _auto_concat_series_chunks(chunk, inputs)
xorbits/_mars/dataframe/merge/concat.py:316: in _auto_concat_series_chunks
    concat = xdf.concat(inputs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

objs = [    size
c2      
a      3
b      5
c      5,     size
c2      
a      8
b      2
c      3,     size
c2      
a      4
b      7
c      2,     size
c2      
a      3
b      2
c      4]
axis = 0, join = 'outer', ignore_index = False, sort = None

    def concat(objs, axis=0, join="outer", ignore_index=False, sort=None):
        """Concatenate DataFrames, Series, or Indices row-wise.

        Parameters
        ----------
        objs : list of DataFrame, Series, or Index
        axis : {0/'index', 1/'columns'}, default 0
            The axis to concatenate along.
        join : {'inner', 'outer'}, default 'outer'
            How to handle indexes on other axis (or axes).
        ignore_index : bool, default False
            Set True to ignore the index of the *objs* and provide a
            default range index instead.
        sort : bool, default False
            Sort non-concatenation axis if it is not already aligned.

        Returns
        -------
        A new object of like type with rows from each object in ``objs``.

        Examples
        --------
        Combine two ``Series``.

        >>> import cudf
        >>> s1 = cudf.Series(['a', 'b'])
        >>> s2 = cudf.Series(['c', 'd'])
        >>> s1
        0    a
        1    b
        dtype: object
        >>> s2
        0    c
        1    d
        dtype: object
        >>> cudf.concat([s1, s2])
        0    a
        1    b
        0    c
        1    d
        dtype: object

        Clear the existing index and reset it in the
        result by setting the ``ignore_index`` option to ``True``.

        >>> cudf.concat([s1, s2], ignore_index=True)
        0    a
        1    b
        2    c
        3    d
        dtype: object

        Combine two DataFrame objects with identical columns.

        >>> df1 = cudf.DataFrame([['a', 1], ['b', 2]],
        ...                    columns=['letter', 'number'])
        >>> df1
          letter  number
        0      a       1
        1      b       2
        >>> df2 = cudf.DataFrame([['c', 3], ['d', 4]],
        ...                    columns=['letter', 'number'])
        >>> df2
          letter  number
        0      c       3
        1      d       4
        >>> cudf.concat([df1, df2])
          letter  number
        0      a       1
        1      b       2
        0      c       3
        1      d       4

        Combine DataFrame objects with overlapping columns and return
        everything. Columns outside the intersection will
        be filled with ``null`` values.

        >>> df3 = cudf.DataFrame([['c', 3, 'cat'], ['d', 4, 'dog']],
        ...                    columns=['letter', 'number', 'animal'])
        >>> df3
          letter  number animal
        0      c       3    cat
        1      d       4    dog
        >>> cudf.concat([df1, df3], sort=False)
          letter  number animal
        0      a       1   <NA>
        1      b       2   <NA>
        0      c       3    cat
        1      d       4    dog

        Combine ``DataFrame`` objects with overlapping columns
        and return only those that are shared by passing ``inner`` to
        the ``join`` keyword argument.

        >>> cudf.concat([df1, df3], join="inner")
          letter  number
        0      a       1
        1      b       2
        0      c       3
        1      d       4

        Combine ``DataFrame`` objects horizontally along the
        x axis by passing in ``axis=1``.

        >>> df4 = cudf.DataFrame([['bird', 'polly'], ['monkey', 'george']],
        ...                    columns=['animal', 'name'])
        >>> df4
           animal    name
        0    bird   polly
        1  monkey  george
        >>> cudf.concat([df1, df4], axis=1)
          letter  number  animal    name
        0      a       1    bird   polly
        1      b       2  monkey  george
        """
        # TODO: Do we really need to have different error messages for an empty
        # list and a list of None?
        if not objs:
            raise ValueError("No objects to concatenate")

        objs = [obj for obj in objs if obj is not None]

        if not objs:
            raise ValueError("All objects passed were None")

        # Return for single object
        if len(objs) == 1:
            obj = objs[0]

            if ignore_index:
                if axis == 1:
                    result = cudf.DataFrame._from_data(
                        data=obj._data.copy(deep=True),
                        index=obj.index.copy(deep=True),
                    )
                    # The DataFrame constructor for dict-like data (such as the
                    # ColumnAccessor given by obj._data here) will drop any columns
                    # in the data that are not in `columns`, so we have to rename
                    # after construction.
                    result.columns = pd.RangeIndex(len(obj._data.names))
                elif axis == 0:
                    if isinstance(obj, cudf.Series):
                        result = cudf.Series._from_data(
                            data=obj._data.copy(deep=True),
                            index=cudf.RangeIndex(len(obj)),
                        )
                    elif isinstance(obj, pd.Series):
                        result = cudf.Series(
                            data=obj,
                            index=cudf.RangeIndex(len(obj)),
                        )
                    else:
                        result = cudf.DataFrame._from_data(
                            data=obj._data.copy(deep=True),
                            index=cudf.RangeIndex(len(obj)),
                        )
            else:
                if axis == 0:
                    result = obj.copy()
                else:
                    data = obj._data.copy(deep=True)
                    if isinstance(obj, cudf.Series) and obj.name is None:
                        # If the Series has no name, pandas renames it to 0.
                        data[0] = data.pop(None)
                    result = cudf.DataFrame._from_data(data)

            return result.sort_index(axis=axis) if sort else result

        # Retrieve the base types of `objs`. In order to support sub-types
        # and object wrappers, we use `isinstance()` instead of comparing
        # types directly
        typs = set()
        for o in objs:
            if isinstance(o, cudf.MultiIndex):
                typs.add(cudf.MultiIndex)
            elif isinstance(o, cudf.BaseIndex):
                typs.add(type(o))
            elif isinstance(o, cudf.DataFrame):
                typs.add(cudf.DataFrame)
            elif isinstance(o, cudf.Series):
                typs.add(cudf.Series)
            else:
>               raise TypeError(f"cannot concatenate object of type {type(o)}")
E               TypeError: cannot concatenate object of type <class 'pandas.core.frame.DataFrame'>

/home/lichengjie/miniconda3/lib/python3.9/site-packages/cudf/core/reshape.py:294: TypeError
```

The reason why this test passes in a CPU-only environment is that cudf is None in a CPU-only environment, so the judgment here in ``line 312  _mars/dataframe/merge/concat.py`` coincidentally returns pandas. Actually, this is a bug.

--- a/python/xorbits/_mars/dataframe/groupby/aggregation.py
+++ b/python/xorbits/_mars/dataframe/groupby/aggregation.py
@@ -1031,9 +1031,6 @@ class DataFrameGroupByAgg(DataFrameOperand, DataFrameOperandMixin):
                     if gpu and agg_func == "size"
                     else input_obj.agg(agg_func)
                 )
-                if result.ndim == 1:
-                    # when agg_func == size, agg only returns one single series.
-                    result = result.to_frame(agg_func)
             else:
                 result = input_obj.agg([agg_func])
                 result.columns = result.columns.droplevel(-1)

